pkg/devfile/adapters/common/command.go

```diff
@@ -134,17 +147,32 @@ func GetRunCommand(data data.DevfileData, devfileRunCmd string) (runCommand comm
 // ValidateAndGetPushDevfileCommands validates the build and the run command,
 // if provided through odo push or else checks the devfile for devBuild and devRun.
 // It returns the build and run commands if its validated successfully, error otherwise.
-func ValidateAndGetPushDevfileCommands(data data.DevfileData, devfileBuildCmd, devfileRunCmd string) (pushDevfileCommands []common.DevfileCommand, err error) {
+func ValidateAndGetPushDevfileCommands(data data.DevfileData, devfileInitCmd, devfileBuildCmd, devfileRunCmd string) (pushDevfileCommands []common.DevfileCommand, err error) {
```

Comment:
I would suggest refactoring all the `Get*Command` to return a `*common.DevfileCommand` as we depend so highly on them to not be empty, checking if the command is `nil` or not will remove most of `reflect` code and increase readability.

Reply:
not sure if it should be done as part of this PR, I would settle for the quickest way.

Reply:
We did notice that `reflect.DeepEqual` introduced more complexity. That is a good idea to solve it, but I would recommend doing it in another PR for all the commands.